index.js

```diff
@@ -17,7 +17,13 @@ function fastifyMysql (fastify, options, next) {
 
     const client = connectionType !== 'connection' ? db.pool : db.connection
 
-    fastify.addHook('onClose', (fastify, done) => client.end(done))
+    if (usePromise) {
+      fastify.addHook('onClose', async (fastify, done) => {
+        await client.end(); done()
```

Comment:
I don't think the call to `done()` is needed. Could you remove?

Reply:
I can't figure out how to test this. Any suggestions on an approach and I will make the changes.

Reply:
The problem here is that we should use the sync or async approach based on the `usePromise` option:

if it is true the hook should be:

```
fastify.addHook('onClose', async (fastify) => {
  await client.end();
})
```

if it is falsy the hook should manage the `done`

We can merge those in:

```
fastify.addHook('onClose', (fastify, done) => {
  const end = client.end()
  if (end && typeof end.then === 'function' ){
    end.then(done, done)
  } else {
    done()
  }
})
```